lib/http.js

```diff
@@ -3,44 +3,52 @@ const callingAppName = require(`${process.cwd()}/package.json`).name;
 const util = require("util");
 const config = require("exp-config");
 const {buildLogger} = require("lu-logger");
-const request = require("request").defaults({json: true});
+const axios = require("axios");
 
 const backends = buildBackends();
 
-function performRequest(method, params) {
-  const logger = buildLogger({
-    meta: {correlationId: params.correlationId, requesterName: callingAppName, routingKey: params.routingKey}
-  });
-
-  return new Promise((resolve, reject) => {
-    const url = `${params.baseUrl || config.proxyUrl}${params.path}`;
-    logger.info(`HTTP ${method}, ${url}, params: ${JSON.stringify(params)}`);
-
-    const opts = {
-      method,
-      url,
-      qs: params.qs,
-      headers: buildHeaders(params, params.headers || {})
-    };
-
-    if (params.body) {
-      opts.body = params.body;
-    }
-    if (params.timeout) {
-      opts.timeout = params.timeout;
-    }
-
-    request(opts, (err, response, body) => {
-      if (err) {
-        logger.warning("HTTP %s:%s yielded %s (error", method, url, response && response.statusCode, err, body, ")");
-        return reject(err);
-      }
+async function performRequest(method, params) {
+  const logger = buildLogger(params.correlationId, params.key);
+  const url = `${params.baseUrl || config.proxyUrl}${params.path}`;
+  logger.info(`HTTP ${method}, ${url}, params: ${JSON.stringify(params)}`);
 
-      logger.info(`HTTP response for ${method} ${url}, ${response.statusCode}, ${JSON.stringify(response.body)}`);
+  // const authHeaders = await gcpAuthHeaders(params.audience || params.baseUrl);
+  const opts = {
+    method,
+    params: {...params.qs, ...params.query},
+    headers: buildHeaders(params, {...(params.headers || {})}),
+    validateStatus: function () {
```

Comment:
A little tighter. Here and further down.

```suggestion
validateStatus: () => true, // do not let axios validate status, we do that ourselves
```